web/docs/develop/architecture/plugins.md

```diff
@@ -24,64 +24,76 @@ static plugins. Because they are always built, we call them *builtins*.
-
-### Language Frontend
-
-A language frontend plugin adds an alternative parser for a query expression.
-This plugin allows for replacing the query *frontend* while using Tenzir as
-*backend* execution engine.
-
-For example, you could write a SQL plugin that takes an expression like
-`SELECT * FROM zeek.conn WHERE id.orig_h = "1.2.3.4"` and executes it on
-historical data or runs it as live query.
+```mermaid
+flowchart TB
+subgraph System
+  Command
+  click Command "#command"
+  Component
+  click Component "#component"
+end
+subgraph Storage
+  StoreActor
+  click StoreActor "#store"
+  Store
+  click Store "#store"
+end
+subgraph Format
+  Parser
+  click Parser "#parser"
+  Printer
+  click Printer "#printer"
+end
+subgraph Connector
+  Loader
+  click Loader "#loader"
+  Saver
+  click Saver "#saver"
+end
+subgraph Pipeline
+  Operator
+  click Operator "#operator"
+  AggregationFunction
+  click AggregationFunction "#aggregation-function"
+end
+```
 
 ### Operator
 
 The pipeline plugin adds a new [pipeline
 operator](../../understand/operators/README.md) that users can reference in a
 [pipeline definition](../../understand/pipelines.md).
 
+### Aggregation Function
+
+The aggregation function plugin adds a new [aggregation
+function](../../understand/operators/transformations/summarize.md#aggregate-functions)
+for the `summarize` pipeline operator that performs an incremental aggregation
+over a set of grouped input values of a single type.
+
 ### Loader
 
-The loader plugin defines the input side of a connector for use in the `from
+The loader plugin defines the input side of a
+[connector](../../understand/connectors/) for use in the `from
 CONNECTOR read FORMAT` and `load CONNECTOR` operators.
 
 ### Saver
 
-The saver plugin defines the output side of a connector for use in the `write
-FORMAT to CONNECTOR` and `save CONNECTOR` operators.
+The saver plugin defines the output side of a
+[connector](../../understand/connectors/) for use in the `write FORMAT to
+CONNECTOR` and `save CONNECTOR` operators.
 
 ### Parser
 
-The parser plugin defines the input side of a format for use in the `from
-CONNECTOR read FORMAT` and `parse FORMAT` operators.
+The parser plugin defines the input side of a
+[format](../../understand/formats/) for use in the `from CONNECTOR read FORMAT`
+and `parse FORMAT` operators.
 
 ### Printer
 
-The parser plugin defines the output side of a format for use in the `write
-FORMAT to CONNECTOR` and `print FORMAT` operators.
-
-### Aggregation Function
-
-The aggregation function plugin adds a new [aggregation
-function](../../understand/operators/transformations/summarize.md#aggregate-functions)
-for the `summarize` pipeline operator that performs an incremental aggregation
-over ajj set of grouped input values of a single type.
+The parser plugin defines the output side of a
+[format](../../understand/formats/) for use in the `write FORMAT to CONNECTOR`
+and `print FORMAT` operators.
 
 ### Store
```

Comment:
This should mention that every store plugin is implicitly also a parser and printer plugin.